tree/ntuple/v7/src/RPageSinkBuf.cxx

```diff
@@ -82,3 +88,55 @@ void ROOT::Experimental::Detail::RPageSinkBuf::ReleasePage(RPage &page)
 {
    fInnerSink->ReleasePage(page);
 }
+
+namespace {
+   using namespace ROOT::Experimental::Detail;
+   struct RPageZipItem {
+      RPage fPage;
+      RNTupleCompressor fCompressor;
+      RPageStorage::RSealedPage fSealedPage;
+      explicit RPageZipItem(RPage page)
+         : fPage(page), fCompressor(RNTupleCompressor(fPage.GetSize())) {}
+   };
+} // anonymous namespace
+
+void ROOT::Experimental::Detail::RPageSinkBuf::ParallelClusterZip(
+   ROOT::Experimental::NTupleSize_t nEntries)
+{
+   // TODO(max) add timers like in RPageSourceFile::UnzipClusterImpl
+   R__ASSERT(fTaskScheduler);
+   fTaskScheduler->Reset();
+   std::vector<std::vector<RPageZipItem>> zipItems;
+   for (auto &col : fBufferedColumns) {
+      std::vector<RPageZipItem> zipCol;
+      for (const auto &page : col.DrainBufferedPages()) {
+         zipCol.push_back(RPageZipItem(page));
+      }
+      zipItems.push_back(std::move(zipCol));
+   }
+   // Thread safety: Each task works on a distinct RPageZipItem `zi`.
+   // Task (i,j) seals RPage (i,j) -- zi.fPage -- using a scratch buffer
+   // owned by zi.fCompressor.
+   for (std::size_t i = 0; i < fBufferedColumns.size(); i++) {
+      for (std::size_t j = 0; j < zipItems.at(i).size(); j++) {
+         fTaskScheduler->AddTask([this, &zipItems, i, j] {
+            auto &zi = zipItems.at(i).at(j);
+            zi.fSealedPage = SealPage(zi.fPage,
+               *fBufferedColumns.at(i).GetHandle().fColumn->GetElement(),
+               fOptions.GetCompression(), zi.fCompressor
+            );
+         });
+      }
+   }
+   fTaskScheduler->Wait();
+
+   for (std::size_t i = 0; i < fBufferedColumns.size(); i++) {
+      for (auto &zi : zipItems.at(i)) {
+         fInnerSink->CommitSealedPage(
+            fBufferedColumns.at(i).GetHandle().fId, zi.fSealedPage
```

Comment:
I hit a confusing bug here when I wrote `fBufferedColumns.at(i).GetHandle()` instead (not passing the actual column id but the handle). Because `RColumnHandle` has a conversion to bool, the handle was converted to bool, then to `DescriptorId_t` because it's a integer typedef. So every page was being added to column 1 (true, because each column handle was valid).

The two `CommitPage` signatures are: 
```cpp
virtual RClusterDescriptor::RLocator CommitPageImpl(ColumnHandle_t columnHandle, const RPage &page) = 0;
virtual RClusterDescriptor::RLocator CommitSealedPageImpl(DescriptorId_t columnId,
                                                          const RPageStorage::RSealedPage &sealedPage) = 0;
```

Reply:
Bool conversion was made explicit in 5ad1bce, so this mistake is now a compile error: 
```
root/tree/ntuple/v7/src/RPageSinkBuf.cxx:136:45: error: cannot convert ‘const ColumnHandle_t’ {aka ‘const ROOT::Experimental::Detail::RPageStorage::RColumnHandle’} to ‘ROOT::Experimental::DescriptorId_t’ {aka ‘long unsigned int’}
  136 |             fBufferedColumns.at(i).GetHandle(), zi.fSealedPage
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
      |                                             |
      |                                             const ColumnHandle_t {aka const ROOT::Experimental::Detail::RPageStorage::RColumnHandle}
```

Reply:
Thanks, making the bool operator explicit is a very useful improvement!